docs/guides/docs/intro-to-sway/typescript-sdk.mdx

```diff
@@ -0,0 +1,350 @@
+  commentType="//"
+  lang="tsx"
+/>
+
+This component will take the contract we made in `App.tsx` as a prop, so let's create an interface for the component.
+
+<CodeImport
+  file="../../../intro-to-sway/frontend/src/components/ListItem.tsx"
+  comment="fe_list_items_interface"
+  commentType="//"
+  lang="tsx"
+/>
+
+We can set up the template for the function like this.
+
+<CodeImport
+  file="../../../intro-to-sway/frontend/src/components/ListItem.tsx"
+  comment="fe_list_items_function"
+  commentType="//"
+  lang="tsx"
+/>
+
+
+To list an item, we'll create a form where the user can input the metadata string and price for the item they want to list. 
+Let's start by adding some state variables for the `metadata` and `price`. We can also add a `status` variable to track the submit status.
+
+<CodeImport
+  file="../../../intro-to-sway/frontend/src/components/ListItem.tsx"
+  comment="fe_list_items_state_variables"
+  commentType="//"
+  lang="tsx"
+/>
+
+Under the heading, add the code below for the form:
+
+<CodeImport
+  file="../../../intro-to-sway/frontend/src/components/ListItem.tsx"
+  comment="fe_list_items_form"
+  commentType="{/*"
+  lang="tsx"
+/>
+
+Finally, we need to add the `handleSubmit` function. 
+We can use the contract prop to call the `list_item` function and pass in the `price` and `metadata` from the form.
+
+<CodeImport
+  file="../../../intro-to-sway/frontend/src/components/ListItem.tsx"
+  comment="fe_list_items_handle_submit"
+  commentType="//"
+  lang="tsx"
+/>
+
+Now that we have this component, let's add it to our `App.tsx` file and try it out.
+
+Import the `ListItem` component at the top of the file.
+Then, replace where it says `Connected!` with the code below:
+
+<CodeImport
+  file="../../../intro-to-sway/frontend/src/App.tsx"
+  comment="fe_items_contract"
+  commentType="{/*"
+  lang="tsx"
+/>
+
+Now, try listing an item to make sure this works. 
+You should see the message `Item successfully listed!`.
+
+### Show All Items
+
+Next, let's create a new file called `AllItems.tsx` in the `components` folder.
+
+Copy and paste the template code below for this component:
+
+<CodeImport
+  file="../../../intro-to-sway/frontend/src/components/AllItems.tsx"
+  comment="fe_all_items_template"
+  commentType="//"
+  lang="tsx"
+/>
+
+Here we can get the item count to see how many items are listed, and then loop through each of them to get the item details.
+
+First, let's create some state variables to store the number of items listed, an array of the item details, and the loading status.
+
+<CodeImport
+  file="../../../intro-to-sway/frontend/src/components/AllItems.tsx"
+  comment="fe_all_items_state_variables"
+  commentType="//"
+  lang="tsx"
+/>
+
+Next, let's fetch the items in a `useEffect` hook.
+Because these are read-only functions, we can simulate a dry-run of the transaction by using the `get` method instead of `call` so the user doesn't have to sign anything.
+
+<CodeImport
```

Comment:
This is a change for the intro-to-sway repo, but there is a new method in the ts sdk to convert to a number, we should use this instead of my work-around `parseFloat(value.format()) * 1_000_000_000;`

Reply:
fixed here https://github.com/FuelLabs/intro-to-sway/pull/21
- [ ] update docs
- [ ] pull in new intro-to-sway here